src/server/pegasus_server_impl.h

```diff
@@ -55,6 +59,8 @@ class pegasus_server_impl : public ::dsn::apps::rrdb_service
     void on_scan(const ::dsn::apps::scan_request &args,
                  ::dsn::rpc_replier<::dsn::apps::scan_response> &reply) override;
     void on_clear_scanner(const int64_t &args) override;
+    void on_detect_hotkey(const ::dsn::apps::hotkey_detect_request &args,
```

Comment:
Should define it as a `rpc_holder`.

```suggestion
    void on_detect_hotkey(::dsn::apps::hotkey_detect_rpc &rpc);
```